tests/test_logger.py

```diff
@@ -434,3 +435,44 @@ def test_ep_buffers_stats_window_size(algo, stats_window_size):
     model.learn(total_timesteps=10)
     assert model.ep_info_buffer.maxlen == stats_window_size
     assert model.ep_success_buffer.maxlen == stats_window_size
+
+
+def test_human_output_format_custom_test_io():
+    class DummyTextIO(TextIOBase):
+        def __init__(self) -> None:
+            super().__init__()
+            self.lines = [[]]
+
+        def write(self, t: str) -> int:
+            self.lines[-1].append(t)
+
+        def flush(self) -> None:
+            self.lines.append([])
+
+        def close(self) -> None:
+            pass
+
+        def get_printed(self) -> str:
+            return "\n".join(["".join(line) for line in self.lines])
+
+    dummy_text_io = DummyTextIO()
+    o = HumanOutputFormat(dummy_text_io)
+    o.write({"key1": "value1", "key2": 42}, {"key1": None, "key2": None})
+    o.write({"key1": "value2", "key2": 43}, {"key1": None, "key2": None})
+    printed = dummy_text_io.get_printed()
+    desired_printed = """-----------------
+| key1 | value1 |
+| key2 | 42     |
+-----------------
+
+-----------------
+| key1 | value2 |
+| key2 | 43     |
+-----------------
+
+"""
+
+    assert printed == desired_printed
+
+
+test_human_output_format_custom_test_io()
```

Comment:
```suggestion
```
you can use `pytest -v tests/ -k human_output_format_custom_test_io` if you want to try a single test

Reply:
It's to use the debugger 😉